stonesoup/models/measurement/nonlinear.py

```diff
@@ -965,3 +967,192 @@ def rvs(self, num_samples=1, **kwargs) -> Union[StateVector, StateVectors]:
+
+    * :math:`h` is a non-linear model function of the form:
+
+    .. math::
+
+      h(\vec{x}_t,\vec{v}_t) = \begin{bmatrix}
+                asin(\mathcal{z}/\sqrt{\mathcal{x}^2 + \mathcal{y}^2 +\mathcal{z}^2}) \\
+                atan2(\mathcal{y},\mathcal{x}) \\
+                \sqrt{\mathcal{x}^2 + \mathcal{y}^2 + \mathcal{z}^2} \\
+                (x\dot{x} + y\dot{y} + z\dot{z})/\sqrt{x^2 + y^2 + z^2}
+                \end{bmatrix} + \vec{v}_t
+
+    * :math:`\vec{v}_t` is Gaussian distributed with covariance :math:`R`, i.e.:
+
+    .. math::
+
+      \vec{v}_t \sim \mathcal{N}(0,R)
+
+    .. math::
+
+      R = \begin{bmatrix}
+            \sigma_{\theta}^2 & 0 & 0 & 0\\
+            0 & \sigma_{\phi}^2 & 0 & 0\\
+            0 & 0 & \sigma_{r}^2 & 0\\
+            0 & 0 & 0 & \sigma_{\dot{r}}^2
+            \end{bmatrix}
+
+    The range and range rate are binned using
+
+    .. math::
+
+        x = floor(x/\Delta x)*\Delta x
+
+    The :py:attr:`mapping` property of the model is a 3 element vector, \
+    whose first (i.e. :py:attr:`mapping[0]`), second (i.e. \
+    :py:attr:`mapping[1]`) and third (i.e. :py:attr:`mapping[2]`) elements \
+    contain the state index of the :math:`x`, :math:`y` and :math:`z`  \
+    coordinates, respectively.
+
+    The :py:attr:`velocity_mapping` property of the model is a 3 element vector, \
+    whose first (i.e. :py:attr:`velocity_mapping[0]`), second (i.e. \
+    :py:attr:`velocity_mapping[1]`) and third (i.e. :py:attr:`velocity_mapping[2]`) elements \
+    contain the state index of the :math:`\dot{x}`, :math:`\dot{y}` and :math:`\dot{z}`  \
+    coordinates, respectively.
+
+    Note
+    ----
+    This class implementation assuming at 3D cartesian space, it therefore \
+    expects a 6D state space.
+    """
+
+    range_res = Property(float, doc="Size of the range bins")
+    range_rate_res = Property(float, doc="Size of the velocity bins")
+
+    @property
+    def ndim_meas(self):
+        return 4
+
+    def function(self, state, noise=False, **kwargs):
+        r"""Model function :math:`h(\vec{x}_t,\vec{v}_t)`
+
+        Parameters
+        ----------
+        state: :class:`~.StateVector`
+            An input state vector for the target
+
+        noise: :class:`numpy.ndarray` or bool
+            An externally generated random process noise sample (the default is
+            `False`, in which case no noise will be added and no binning takes place
+            if 'True', the output of :meth:`~.Model.rvs` is added and the
+            range and range rate are binned)
+
+        Returns
+        -------
+        :class:`numpy.ndarray` of shape (:py:attr:`~ndim_state`, 1)
+            The model function evaluated given the provided time interval.
+        """
+
+        out = super().function(state, noise, **kwargs)
+
+        if isinstance(noise, bool) or noise is None:
+            if noise:
+                out[0] = np.floor(out[0] / self.range_res) * self.range_res
```

Comment:
If there is no `noise`, or `noise` is supplied then don't bin. Is that logical? Why would whether you bin depend on whether noise is added?

Reply:
It's important for the PDF. this is because "state2" is passed into self.function with noise = false. Then that's used as the mean in multivariance.norm.cdf (pdf for other models). If it was bin it would move state2 which would move the center of the normal distribution.

Reply:
Would the check be better as `if not noise == False` then? Why check for if it is None? Again, there might be a good answer. 

I would also add the justification for this to either the docstring or a comment.

Reply:
I'm not sure why it's done the way it is but it's what the other measurement models use. I'll ask Rich

Reply:
If I remember rightly, this is because combination of `0 == False`, and `if not np.array([[1], [2], [3]]) == False:` would raise _ValueError_.

Reply:
Would `if noise is False` work? Seems to salve those two issues.

Reply:
To save a line, could you do: 'if isinstance(noise, bool) and noise:' ?